woocommerce/woocommerce-ios
# Product Selector: Add filter functionality

Part of #6489 


### Description

This PR continues the customization of the product selector to add a filter bar on top of the product list:
- Add a bar with two new buttons: Select All and Filter.
  - The Select All button should only be visible when multiple selections are enabled and when there are results present on the view.
  - The Filter is available when the `showsFilter` configuration is enabled for the product selector.
- Handle filter functionality:
  - Wrap `FilterListViewController` in a SwiftUI view to be presented from the product selector.
  - Update `ProductSelectorViewModel` to update the product results controller when filters are updated.
  - Update the title of the filter button when the number of active filters changes.
- Enable filter for `AddEditCoupon` and `CouponRestrictions`.

### Testing instructions

Testing coupon edit and coupon restrictions:
- Make sure that your test store has coupons enabled and has set up at least one coupon in WP Admin > Marketing > Coupons.
- Open the app, make sure that coupon management is enabled in Menu > Settings > Experimental Features > Coupon Management.
- Navigate to Menu > Coupons and select a coupon.
- Tap the ellipsis button > Edit Coupon > Select the All Products/Edit Products button.
- Notice that 2 buttons Select All and Filter are present.
- Tap the Filter button, update the filters to your liking, and apply the filter. The product list should show only the matching items, and the Filter button should display the correct number of active filters.
- If no result is found, the Filter button should still be present while the Select All button is not.

Please feel free to test the product selector on the New Order screen to make sure that single selection still works as before.

### Screenshots


https://user-images.githubusercontent.com/5533851/164704470-f7737cd8-e7fd-4c62-ac7f-8a3906e651be.mp4



---
